cmd/gitopsUpdateDeployment_test.go

```diff
@@ -42,75 +42,225 @@ func TestBuildRegistryPlusImage(t *testing.T) {
-		assert.EqualError(t, err, "failed to apply kubectl command: registry URL could not be extracted: invalid registry url")
+		err := runGitopsUpdateDeployment(validConfiguration, &gitOpsExecRunnerMock{}, gitUtils, &filesMock{})
+		assert.EqualError(t, err, "repository could not get prepared: failed to plain clone repository: error on clone")
 	})
 
-	t.Run("error on plane clone", func(t *testing.T) {
-		var configuration = &gitopsUpdateDeploymentOptions{
-			BranchName:            "main",
-			CommitMessage:         "This is the commit message",
-			ServerURL:             "https://github.com",
-			Username:              "admin3",
-			Password:              "validAccessToken",
-			FilePath:              "dir1/dir2/depl.yaml",
-			ContainerName:         "myContainer",
-			ContainerRegistryURL:  "https://myregistry.com/registry/containers",
-			ContainerImageNameTag: "myFancyContainer:1337",
-		}
+	t.Run("error on change branch", func(t *testing.T) {
+		gitUtils := &gitUtilsMock{failOnChangeBranch: true}
+
+		err := runGitopsUpdateDeployment(validConfiguration, &gitOpsExecRunnerMock{}, gitUtils, &filesMock{})
+		assert.EqualError(t, err, "repository could not get prepared: failed to change branch: error on change branch")
+	})
+
+	t.Run("error on commit changes", func(t *testing.T) {
+		gitUtils := &gitUtilsMock{failOnCommit: true}
+
+		err := runGitopsUpdateDeployment(validConfiguration, &gitOpsExecRunnerMock{}, gitUtils, &filesMock{})
+		assert.EqualError(t, err, "failed to commit and push changes: committing changes failed: error on commit")
+	})
+
+	t.Run("error on push commits", func(t *testing.T) {
+		gitUtils := &gitUtilsMock{failOnPush: true}
 
-		err := runGitopsUpdateDeployment(configuration, nil, &gitUtilsMockErrorClone{}, piperutils.Files{})
-		assert.EqualError(t, err, "failed to plain clone repository: error on clone")
+		err := runGitopsUpdateDeployment(validConfiguration, &gitOpsExecRunnerMock{}, gitUtils, &filesMock{})
+		assert.EqualError(t, err, "failed to commit and push changes: pushing changes failed: error on push")
 	})
 
 	t.Run("error on temp dir creation", func(t *testing.T) {
+		fileUtils := &filesMock{failOnCreation: true}
+
+		err := runGitopsUpdateDeployment(validConfiguration, &gitOpsExecRunnerMock{}, &gitUtilsMock{}, fileUtils)
+		assert.EqualError(t, err, "failed to create temporary directory: error appeared")
+	})
+
+	t.Run("error on file write", func(t *testing.T) {
+		fileUtils := &filesMock{failOnWrite: true}
+
+		err := runGitopsUpdateDeployment(validConfiguration, &gitOpsExecRunnerMock{}, &gitUtilsMock{}, fileUtils)
+		assert.EqualError(t, err, "failed to write file: error appeared")
+	})
+
+	t.Run("error on temp dir deletion", func(t *testing.T) {
+		fileUtils := &filesMock{failOnDeletion: true}
+
+		err := runGitopsUpdateDeployment(validConfiguration, &gitOpsExecRunnerMock{}, &gitUtilsMock{}, fileUtils)
+		assert.NoError(t, err)
+		_ = piperutils.Files{}.RemoveAll(fileUtils.path)
```

Comment:
This is strange. Why is this necessary?

Reply:
There were tests failing (https://github.com/SAP/jenkins-library/runs/1342216341?check_suite_focus=true) because this directory still exists, but I guess this is not the case anymore since the temporary directory will be used in the mocks glob method at the commit part.